src/scaffoldmaker/meshtypes/meshtype_3d_bladderurethra1.py

```diff
@@ -813,14 +814,68 @@ def generateBaseMesh(cls, region, options):
                             ureterMeshGroup, bladderMeshGroup)
 
         # Apex annotation point
-        nodeIdentifier = nextNodeIdentifier
+        nodeIdentifier = nextNodeIdentifier + elementsCountAroundUreter * 4 * 2 + elementsCountAroundUreter * 2 * 2 * elementsCountUreterRadial
         idx = 1
         element1 = mesh.findElementByIdentifier(idx)
         markerPoint = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
         cache.setNode(markerPoint)
         markerName.assignString(cache, 'Apex of urinary bladder')
         markerLocation.assignMeshLocation(cache, element1, [0.0, 0.0, 1.0])
 
+        # Markers for ureters
+        if includeUreter:
+            # Left ureter
+            nodeIdentifier += 1
+            idx1 = elementsCountAlong * elementsCountAround + elementsCountAroundUreter
+            element1 = mesh.findElementByIdentifier(idx1)
+            markerUreter1Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
+            cache.setNode(markerUreter1Point)
+            markerName.assignString(cache, 'ureter junction of bladder trigone')
+            markerLocation.assignMeshLocation(cache, element1, [0.0, 1.0, 1.0])
+            # Right ureter
+            nodeIdentifier += 1
+            idx2 = elementsCountAlong * elementsCountAround + 2 * elementsCountAroundUreter
+            element2 = mesh.findElementByIdentifier(idx2)
+            markerUreter2Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
+            cache.setNode(markerUreter2Point)
+            markerName.assignString(cache, 'ureter junction of bladder trigone')
+            markerLocation.assignMeshLocation(cache, element2, [0.0, 1.0, 1.0])
+        else:
+            # Left ureter
+            nodeIdentifier += 1
+            idx1 = ureterElementPositionDown * elementsCountAround + ureterElementPositionAround + 1
+            element1 = mesh.findElementByIdentifier(idx1)
+            markerUreter1Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
+            cache.setNode(markerUreter1Point)
+            markerName.assignString(cache, 'ureter junction of bladder trigone')
+            markerLocation.assignMeshLocation(cache, element1, [ureter1Position.xi1, ureter1Position.xi2, 1.0])
+            # Right ureter
+            nodeIdentifier += 1
+            idx2 = ureterElementPositionDown * elementsCountAround + elementsCountAround - ureterElementPositionAround
+            element2 = mesh.findElementByIdentifier(idx2)
+            markerUreter2Point = markerPoints.createNode(nodeIdentifier, markerTemplateInternal)
+            cache.setNode(markerUreter2Point)
+            markerName.assignString(cache, 'ureter junction of bladder trigone')
+            markerLocation.assignMeshLocation(cache, element2, [1-ureter1Position.xi1, ureter1Position.xi2, 1.0])
```

Comment:
I don't like having 4 sets of code to make equivalent points. Prone to maintenance problems.
Can you propose a way to store the points in a structure, e.g. a dict { "name" : "ureter...", "element_id" : idx1, "xi" : [ x, y, z ] }. Then make the 2 points in common code below. I like to see minimum code in special case clauses.
Need to qualify names with left/right.
Do you mean 'ureter junction with bladder trigone'? Is it necessary to specify trigone?